server/src/main/java/com/flipkart/varadhi/web/v1/admin/TopicHandlers.java

```diff
@@ -97,22 +114,43 @@ public void create(RoutingContext ctx) {
         }
 
         Project project = metaStore.getProject(topicResource.getProject());
-        boolean found = metaStore.checkTopicResourceExists(topicResource.getName(), topicResource.getProject());
+        String varadhiTopicName = String.join(NAME_SEPARATOR, projectName, topicResource.getName());
+        boolean found = metaStore.checkVaradhiTopicExists(varadhiTopicName);
```

Comment:
I think we need this check here before calling planDeployment() within the varadhiTopicFactory.get() method. Change done : introduced this check in the service class and called service instead of metastore.